pytorch_lightning/trainer/supporters.py

```diff
@@ -253,7 +291,14 @@ def __init__(self, loaders: Any, mode='min_size'):
                 'max_size_cycle' which stops if the longest loader is exhausted and cycles through the smaller ones.
         """
         self.loaders = loaders
+        # could be multiple datasets, but use self.dataset to follow the name convention in DataLoader
+        datasets = apply_to_collection(self.loaders, Iterable, self.extract_dataset,
+                                       wrong_dtype=(Sequence, Mapping))
+
+        self.dataset = CombinedDataset(datasets)
```

Comment:
what do you need this for?

Reply:
The function `_adjust_batch_size()`, which is used when `Trainer(auto_scale_batch_size=True)`, needs the `dataset` attribute in the data loader. It seems to make sure the adjusted batch size not exceeding the size of the dataset.
https://github.com/PyTorchLightning/pytorch-lightning/blob/5fd1afb38a69f1febbc8529c75efa7fbf9f77dd1/pytorch_lightning/tuner/batch_size_scaling.py#L273-L274

At this time, I add the `dataset` attribute in the CombinedLoader to fit this part. However, if it is not a good choice, maybe we could just disable `_adjust_batch_size()` when there are multiple loaders.

Reply:
I see, I think this should be fine, since you provide a default value for the `getattr`.